# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../cmake/uname.patch
# Copyright (C) 2006 - 2008 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

diff -Nur cmake-2.6.1-orig/Modules/CMakeDetermineSystem.cmake cmake-2.6.1/Modules/CMakeDetermineSystem.cmake
--- cmake-2.6.1-orig/Modules/CMakeDetermineSystem.cmake 2008-08-31 11:22:20.000000000 +0200
+++ cmake-2.6.1/Modules/CMakeDetermineSystem.cmake      2008-08-31 11:25:26.000000000 +0200
@@ -31,16 +31,16 @@
 IF(CMAKE_HOST_UNIX)
   FIND_PROGRAM(CMAKE_UNAME uname /bin /usr/bin /usr/local/bin )
   IF(CMAKE_UNAME)
-    EXEC_PROGRAM(uname ARGS -s OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_NAME)
-    EXEC_PROGRAM(uname ARGS -r OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_VERSION)
+    EXEC_PROGRAM(uname ARGS -s 2>/dev/null | cat OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_NAME)
+    EXEC_PROGRAM(uname ARGS -r 2>/dev/null | cat OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_VERSION)
     IF(CMAKE_HOST_SYSTEM_NAME MATCHES "Linux")
-      EXEC_PROGRAM(uname ARGS -m OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_PROCESSOR
+      EXEC_PROGRAM(uname ARGS -m 2>/dev/null | cat OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_PROCESSOR
         RETURN_VALUE val)
     ELSE(CMAKE_HOST_SYSTEM_NAME MATCHES "Linux")
-      EXEC_PROGRAM(uname ARGS -p OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_PROCESSOR
+      EXEC_PROGRAM(uname ARGS -p 2>/dev/null | cat OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_PROCESSOR
         RETURN_VALUE val)
       IF("${val}" GREATER 0)
-        EXEC_PROGRAM(uname ARGS -m OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_PROCESSOR
+        EXEC_PROGRAM(uname ARGS -m 2>/dev/null | cat OUTPUT_VARIABLE CMAKE_HOST_SYSTEM_PROCESSOR
           RETURN_VALUE val)
       ENDIF("${val}" GREATER 0)
     ENDIF(CMAKE_HOST_SYSTEM_NAME MATCHES "Linux")
